src/ray/raylet/node_manager.cc

```diff
@@ -1308,9 +1327,7 @@ void NodeManager::AssignTask(Task &task) {
   } else {
     RAY_LOG(WARNING) << "Failed to send task to worker, disconnecting client";
     // We failed to send the task to the worker, so disconnect the worker.
-    ProcessClientMessage(worker->Connection(),
-                         static_cast<int64_t>(protocol::MessageType::DisconnectClient),
-                         nullptr);
+    ProcessDisconnectClientMessage(worker->Connection());
```

Comment:
So we do this because it saves a redirection?

Reply:
No, performance here is not important because clients disconnect infrequently. I think this is just a cleaner way of doing it. E.g., `ProcessClientMessage` is for when a client actually sends a message, and no message is being sent here.